basics_of_logging_in_sst-core/Simplified_SST_Logging.md

```diff
@@ -0,0 +1,329 @@
+
+```
+2021-02-26 10:15:12; File=main.cc; Func=main; Line=42; WARNING; Cannot open configuration file
+```
+
+## LOGGING IN SST
+
+Logging within SST is provided by the Output class.  The Output class allows the programmer to customize log messages and has different methods used to print.
+
+### Customizing Log Messages
+
+Log messages can be customized using the following properties.
+
+**Prefix Data.**  Prefix data is a formatted string printed at the beginning of the log message, before the programmer's message.  The following information can be included in the prefix data.
+
+* @f - Name of the file from which the log message is being printed.
+* @l - File line number from which the log message is being printed.
+* @p - Name of the function from which the log message is being printed.
+* @r - MPI rank of the calling process.  Will be empty if there is only a single rank.
+* @R - MPI rank of the calling process.  Will be 0 if there is only a single rank.
+* @i - Calling process thread id.  Will be empty if there is only a single thread.
+* @I - Calling process thread id.  Will be 0 if there is only a single thread.
+* @x - Calling process rank information.  Will be empty if the number of MPI ranks and the number of threads are both 1.  Same as @r:@i.
+* @X - Calling process rank information.  Will be 0.0 if the number of MPI ranks and the number of threads are both 1.  Same as @R:@I.
+* @t - Simulation time.  This is the raw simulation cycle time retrieved from the SST Core.
+
+**Verbose Level.**  The Output class verbose level corresponds to the Syslog Protocol severity level.  When an instance of the output class is instantiated an associated verbose level is assigned and only messages with a higher or equal verbose level (lower or equal numeric value) will be logged.  So, for example, if an instance of the Output class with a verbose level of 10 is instantiated, only messages with verbose levels of 0 - 10 will be logged.  
+
+**Verbose Mask.**  The Output class verbose mask serves as an additional filter on log messages.  Only those log messages whose verbose mask matches the one specified when the Output class was instantiated will be printed.
+
+**Target Output Location.**  The location where log messages are to be sent is specified when the Output class is instantiated.  Available choices are SST::Output::STDERR, SST::Output::STDOUT, SST::Output::NONE, or SST::Output::FILE.  If SST::Output::FILE is specified, the output file name can also be specified.
+
+### Writing Log Messages
+
+The SST Output class provides four different methods to write log messages.
+
+**Output.**  This method works like a standard printf.  By default its messages do not contain the prefix data but can if desired.  
```

Comment:
Calls to output will print unconditionally, regardless of verbose level and flag set.

Also, these functions are all actually lowercase.